test/e2e/framework.go

```diff
@@ -880,7 +895,31 @@ func (data *TestData) deployFlowAggregator(ipfixCollector string) error {
 	if err != nil || rc != 0 {
 		return fmt.Errorf("error when deploying the Flow Aggregator; %s not available on the control-plane Node", flowAggYaml)
 	}
-	if err = data.mutateFlowAggregatorConfigMap(ipfixCollector); err != nil {
+	// clickhouse-ca Secret is created in the flow-visibility Namespace. In order to make it accessible to the Flow Aggregator,
+	// we copy it from Namespace flow-visibility to Namespace flow-aggregator when secureConnection is true.
+	if o.secureConnection {
+		secret, err := data.clientset.CoreV1().Secrets(flowVisibilityNamespace).Get(context.TODO(), flowAggregatorCHSecret, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to get Secret with name %s in Namespace %s: %v", flowAggregatorCHSecret, flowVisibilityNamespace, err)
+		}
+		newSecret := &corev1.Secret{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: corev1.SchemeGroupVersion.String(),
+				Kind:       "Secret",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: flowAggregatorNamespace,
+				Name:      flowAggregatorCHSecret,
+			},
+			Data: secret.Data,
+		}
+		_, err = data.clientset.CoreV1().Secrets(flowAggregatorNamespace).Create(context.TODO(), newSecret, metav1.CreateOptions{})
```

Comment:
for robustness you should call `Update` if the `Create` fails with "already exists"
you may want to delete the secret explicitly at the end of the test, but as long as you delete the `flow-aggregator` Namespace as part of the FA teardown, it is good enough

Reply:
Updated. Thanks.